core/engine/src/native_function.rs

```diff
@@ -391,20 +409,32 @@ fn native_function_construct(
     obj: &JsObject,
     argument_count: usize,
     context: &mut Context,
+) -> JsResult<CallValue> {
+    native_function_construct_inner(
+        &obj.downcast_ref::<NativeFunctionObject>()
+            .expect("the object should be a native function object")
+            .clone(),
+        obj.clone(),
+        argument_count,
+        context,
+    )
+}
+
+pub(crate) fn native_function_construct_inner(
```

Comment:
So there's been some (minor) performance regressions with this PR and I think its related to this, I'm going to assume because its not `[inline]`'d this is now calling 2 functions instead of 1, and because its on a hotpath its actually slowed it down. Ill try adding the inline attribute and see if thats what it is, but leaving this here as a reminder.